From cd60e2f483843f472414f9b901a591119571cc2f Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Thu, 24 Mar 2022 14:47:53 +0100
Subject: [PATCH 0149/1695] clk: stm32mp13: do not reconfigure LSE

If LSE oscillator is already ON, which is the case when returning from
low-power state or if we are on VBAT, it mustn't be reconfigured.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 14bfca5cb6469703aea2bc147f02f10f83dc25b7 clk: stm32mp13: do not reconfigure LSE]
Change-Id: I78ffdd951c0e83fe646a6ca9ee502c999796e3d8
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
---
 core/drivers/clk/clk-stm32mp13.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/drivers/clk/clk-stm32mp13.c b/core/drivers/clk/clk-stm32mp13.c
index bdadcadbd..0a3781657 100644
--- a/core/drivers/clk/clk-stm32mp13.c
+++ b/core/drivers/clk/clk-stm32mp13.c
@@ -777,6 +777,9 @@ static void stm32_enable_oscillator_lse(struct clk_stm32_priv *priv,
 	if (osci->freq == 0U)
 		return;
 
+	if (stm32_gate_is_enabled(osc_data->gate_id))
+		return;
+
 	clk_oscillator_set_bypass(priv, osc_data, osci->digbyp, osci->bypass);
 
 	clk_oscillator_set_drive(priv, osc_data,  osci->drive);
-- 
2.47.2

